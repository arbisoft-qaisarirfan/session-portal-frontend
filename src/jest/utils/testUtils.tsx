import React, { ReactElement } from "react";

import "@testing-library/jest-dom";
import { render, RenderOptions } from "@testing-library/react";

import ThemeProvider from "@/components/theme/theme-provider";
import { Providers } from "@/redux/store/provider";

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, "wrapper">) =>
  render(ui, {
    wrapper: ({ children }) => (
      <Providers>
        <ThemeProvider>{children}</ThemeProvider>
      </Providers>
    ),
    ...options,
  });

export * from "@testing-library/react";

export { customRender };
