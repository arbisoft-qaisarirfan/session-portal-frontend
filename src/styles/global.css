a {
    font-weight: var(--fontWeightNormal);
    color: var(--anchorColor);
    text-decoration: inherit;
}

a:hover {
    color: var(--assistColor);
}

button {
    border-radius: var(--buttonPrimary-border-radius);
    border: var(--buttonPrimary-border);
    padding: var(--buttonPrimary-padding);
    font-size: var(--buttonPrimary-font-size);
    font-weight: var(--buttonPrimary-font-weight);
    font-family: var(--buttonPrimary-font-family);
    background-color: var(--buttonPrimary-background-color);
    cursor: var(--buttonPrimary-cursor);
    transition: var(--buttonPrimary-transition);
}

button:hover {
    border-color: var(--assistColor);
}

button:focus,
button:focus-visible {
    outline: 4px auto -webkit-focus-ring-color;
}

/* Light Mode Styles */
@media (prefers-color-scheme: light) {
    :root {
        color: var(--secondaryColor);
        background-color: var(--whiteColor);
    }

    a:hover {
        color: var(--anchorColorLight);
    }

    button {
        background-color: var(--whiteColor);
    }
}
