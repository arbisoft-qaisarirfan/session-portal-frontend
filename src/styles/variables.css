:root {
    /* TYPOGRAPHY */
    --fontPrimary: Inter, system-ui, Avenir, Helvetica, Arial, sans-serif;
    --fontSizeHeading: 3.2em;
    --fontSizeNormal: 16px;
    --fontWeightSmall: 200px;
    --fontWeightNormal: 400px;
  
    /* COLORS */
    /* TODO: replace these colors with actual colors */
    --assistColor: #508991;
    --assistColorLight: #eef3f4;
    --primaryColor: #004346;
    --whiteColor: #fff;
    --secondaryColor: #172a3a;
    --yellowColor: #f2c94c;
    --formLight: #b5b5b5;
    --ratingStarColor: #f2c94c;
    --color: rgba(255, 255, 255, 0.87);
    --backgroundColor: #242424;
    --anchorColor: #0000FF;
  
    /* Anchor Colors */
    --anchorColorHover: #FF6347;
    --anchorColorLight: #8b8bf6;
  
    /* LINE HEIGHTS */
    --lineHeightHeading: 2;
    --lineHeightNormal: 1.5;
    --lineHeightSmall: 1.15;
  
    /* Button Styles */
    --buttonPrimary-border-radius: 8px;
    --buttonPrimary-border: 1px solid transparent;
    --buttonPrimary-padding: 0.6em 1.2em;
    --buttonPrimary-font-size: 1em;
    --buttonPrimary-font-weight: var(--fontWeightNormal);
    --buttonPrimary-font-family: var(--fontPrimary);
    --buttonPrimary-background-color: var(--backgroundColor);
    --buttonPrimary-cursor: pointer;
    --buttonPrimary-transition: border-color 0.25s;
}