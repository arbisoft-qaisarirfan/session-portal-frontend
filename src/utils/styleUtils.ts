export const shouldForwardProp = (propName: string) => !propName.startsWith("$");
