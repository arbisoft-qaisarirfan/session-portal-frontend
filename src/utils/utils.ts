export function secondsToTime(seconds: number): string {
  const hours = Math.floor(seconds / 3600);
  const minutes = Math.floor((seconds % 3600) / 60);

  const formatNumber = (num: number): string =>
    num.toLocaleString(undefined, {
      useGrouping: false,
      minimumIntegerDigits: 2,
    });

  return `${formatNumber(hours)}:${formatNumber(minutes)}`;
}
