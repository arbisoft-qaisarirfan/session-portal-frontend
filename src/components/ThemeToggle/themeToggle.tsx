import DarkModeIcon from "@mui/icons-material/DarkMode";
import LightModeIcon from "@mui/icons-material/LightMode";
import SettingsBrightnessIcon from "@mui/icons-material/SettingsBrightness";
import Button from "@mui/material/Button";
import ButtonGroup from "@mui/material/ButtonGroup";
import { useColorScheme } from "@mui/material/styles";
import { useTranslation } from "react-i18next";

const ThemeToggle = () => {
  const { mode, setMode } = useColorScheme();

  const { t } = useTranslation("common");

  return (
    <>
      <ButtonGroup size="small" variant="outlined" aria-label="Basic button group">
        <Button
          variant={mode === "light" ? "contained" : undefined}
          onClick={() => setMode("light")}
          startIcon={<LightModeIcon />}
        >
          {t("light")}
        </Button>
        <Button
          variant={mode === "system" ? "contained" : undefined}
          onClick={() => setMode("system")}
          startIcon={<SettingsBrightnessIcon />}
        >
          {t("system")}
        </Button>
        <Button variant={mode === "dark" ? "contained" : undefined} onClick={() => setMode("dark")} startIcon={<DarkModeIcon />}>
          {t("dark")}
        </Button>
      </ButtonGroup>
    </>
  );
};

export default ThemeToggle;
