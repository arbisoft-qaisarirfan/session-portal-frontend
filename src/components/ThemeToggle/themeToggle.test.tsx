import { useColorScheme } from "@mui/material/styles";

import { customRender as render, screen, fireEvent } from "@/jest/utils/testUtils";

import ThemeToggle from "./themeToggle";

jest.mock("@mui/material/styles", () => ({
  ...jest.requireActual("@mui/material/styles"), // Preserve other exports
  useColorScheme: jest.fn(), // Mock useColorScheme
}));

describe("ThemeToggle", () => {
  const mockSetMode = jest.fn();

  beforeEach(() => {
    // Reset mocks before each test
    jest.clearAllMocks();
  });

  it("renders the light mode button as active when mode is light", () => {
    // Mock the return value of useColorScheme
    (useColorScheme as jest.Mock).mockReturnValue({
      mode: "light",
      setMode: mockSetMode,
    });

    render(<ThemeToggle />);

    // Check that the light mode button is active
    const lightButton = screen.getByRole("button", { name: /light/i });
    expect(lightButton).toHaveClass("MuiButton-contained");

    // Check that the other buttons are not active
    const systemButton = screen.getByRole("button", { name: /system/i });
    expect(systemButton).not.toHaveClass("MuiButton-contained");

    const darkButton = screen.getByRole("button", { name: /dark/i });
    expect(darkButton).not.toHaveClass("MuiButton-contained");
  });

  it("renders the system mode button as active when mode is system", () => {
    // Mock the return value of useColorScheme
    (useColorScheme as jest.Mock).mockReturnValue({
      mode: "system",
      setMode: mockSetMode,
    });

    render(<ThemeToggle />);

    // Check that the system mode button is active
    const systemButton = screen.getByRole("button", { name: /system/i });
    expect(systemButton).toHaveClass("MuiButton-contained");

    // Check that the other buttons are not active
    const lightButton = screen.getByRole("button", { name: /light/i });
    expect(lightButton).not.toHaveClass("MuiButton-contained");

    const darkButton = screen.getByRole("button", { name: /dark/i });
    expect(darkButton).not.toHaveClass("MuiButton-contained");
  });

  it("renders the dark mode button as active when mode is dark", () => {
    // Mock the return value of useColorScheme
    (useColorScheme as jest.Mock).mockReturnValue({
      mode: "dark",
      setMode: mockSetMode,
    });

    render(<ThemeToggle />);

    // Check that the dark mode button is active
    const darkButton = screen.getByRole("button", { name: /dark/i });
    expect(darkButton).toHaveClass("MuiButton-contained");

    // Check that the other buttons are not active
    const lightButton = screen.getByRole("button", { name: /light/i });
    expect(lightButton).not.toHaveClass("MuiButton-contained");

    const systemButton = screen.getByRole("button", { name: /system/i });
    expect(systemButton).not.toHaveClass("MuiButton-contained");
  });

  it("calls setMode with 'light' when the light mode button is clicked", () => {
    // Mock the return value of useColorScheme
    (useColorScheme as jest.Mock).mockReturnValue({
      mode: "system",
      setMode: mockSetMode,
    });

    render(<ThemeToggle />);

    // Simulate a click on the light mode button
    const lightButton = screen.getByRole("button", { name: /light/i });
    fireEvent.click(lightButton);

    // Verify that setMode was called with 'light'
    expect(mockSetMode).toHaveBeenCalledWith("light");
  });

  it("calls setMode with 'system' when the system mode button is clicked", () => {
    // Mock the return value of useColorScheme
    (useColorScheme as jest.Mock).mockReturnValue({
      mode: "dark",
      setMode: mockSetMode,
    });

    render(<ThemeToggle />);

    // Simulate a click on the system mode button
    const systemButton = screen.getByRole("button", { name: /system/i });
    fireEvent.click(systemButton);

    // Verify that setMode was called with 'system'
    expect(mockSetMode).toHaveBeenCalledWith("system");
  });

  it("calls setMode with 'dark' when the dark mode button is clicked", () => {
    // Mock the return value of useColorScheme
    (useColorScheme as jest.Mock).mockReturnValue({
      mode: "light",
      setMode: mockSetMode,
    });

    render(<ThemeToggle />);

    // Simulate a click on the dark mode button
    const darkButton = screen.getByRole("button", { name: /dark/i });
    fireEvent.click(darkButton);

    // Verify that setMode was called with 'dark'
    expect(mockSetMode).toHaveBeenCalledWith("dark");
  });

  test("should snapshot test for ThemeToggle", () => {
    const { asFragment } = render(<ThemeToggle />);
    expect(asFragment()).toMatchSnapshot();
  });
});
