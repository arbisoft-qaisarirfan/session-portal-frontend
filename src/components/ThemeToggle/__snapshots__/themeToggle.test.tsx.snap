// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeToggle should snapshot test for ThemeToggle 1`] = `
<DocumentFragment>
  <div
    aria-label="Basic button group"
    class="MuiButtonGroup-root MuiButtonGroup-outlined MuiButtonGroup-horizontal MuiButtonGroup-colorPrimary css-smlsl6-MuiButtonGroup-root"
    role="group"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButtonGroup-firstButton css-rs20bt-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeSmall css-1tjwu2m-MuiButton-startIcon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-4uxqju-MuiSvgIcon-root"
          data-testid="LightModeIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5M2 13h2c.55 0 1-.45 1-1s-.45-1-1-1H2c-.55 0-1 .45-1 1s.45 1 1 1m18 0h2c.55 0 1-.45 1-1s-.45-1-1-1h-2c-.55 0-1 .45-1 1s.45 1 1 1M11 2v2c0 .55.45 1 1 1s1-.45 1-1V2c0-.55-.45-1-1-1s-1 .45-1 1m0 18v2c0 .55.45 1 1 1s1-.45 1-1v-2c0-.55-.45-1-1-1s-1 .45-1 1M5.99 4.58c-.39-.39-1.03-.39-1.41 0-.39.39-.39 1.03 0 1.41l1.06 1.06c.39.39 1.03.39 1.41 0s.39-1.03 0-1.41zm12.37 12.37c-.39-.39-1.03-.39-1.41 0-.39.39-.39 1.03 0 1.41l1.06 1.06c.39.39 1.03.39 1.41 0 .39-.39.39-1.03 0-1.41zm1.06-10.96c.39-.39.39-1.03 0-1.41-.39-.39-1.03-.39-1.41 0l-1.06 1.06c-.39.39-.39 1.03 0 1.41s1.03.39 1.41 0zM7.05 18.36c.39-.39.39-1.03 0-1.41-.39-.39-1.03-.39-1.41 0l-1.06 1.06c-.39.39-.39 1.03 0 1.41s1.03.39 1.41 0z"
          />
        </svg>
      </span>
      light
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButtonGroup-middleButton css-4zwuhn-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeSmall css-1tjwu2m-MuiButton-startIcon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-4uxqju-MuiSvgIcon-root"
          data-testid="SettingsBrightnessIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M21 3H3c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h18c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2m0 16.01H3V4.99h18zM8 16h2.5l1.5 1.5 1.5-1.5H16v-2.5l1.5-1.5-1.5-1.5V8h-2.5L12 6.5 10.5 8H8v2.5L6.5 12 8 13.5zm4-7c1.66 0 3 1.34 3 3s-1.34 3-3 3z"
          />
        </svg>
      </span>
      system
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButtonGroup-lastButton css-4zwuhn-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeSmall css-1tjwu2m-MuiButton-startIcon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-4uxqju-MuiSvgIcon-root"
          data-testid="DarkModeIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 3c-4.97 0-9 4.03-9 9s4.03 9 9 9 9-4.03 9-9c0-.46-.04-.92-.1-1.36-.98 1.37-2.58 2.26-4.4 2.26-2.98 0-5.4-2.42-5.4-5.4 0-1.81.89-3.42 2.26-4.4-.44-.06-.9-.1-1.36-.1"
          />
        </svg>
      </span>
      dark
    </button>
  </div>
</DocumentFragment>
`;
